### PR TITLE
A small set of parsing / rendering optimizations

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala [hunk 1]
```diff
@@ -51,8 +51,8 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
                             expect100continue: Boolean, hostHeaderPresent: Boolean, closeAfterResponseCompletion: Boolean): HttpMessageParser.StateResult
 
   protected final def initialHeaderBuffer: ListBuffer[HttpHeader] =
-    if (settings.includeTlsSessionInfoHeader && tlsSessionInfoHeader != null) ListBuffer(tlsSessionInfoHeader)
-    else ListBuffer()
+    if (settings.includeTlsSessionInfoHeader && tlsSessionInfoHeader != null) new ListBuffer() += tlsSessionInfoHeader
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala]: hah!

**Comment by jrudolph** [comment]
Build hang for 18 hours...

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by ktoso** [comment]
PLS BUILD

File: akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpResponseRendererFactory.scala [hunk 2]
```diff
@@ -242,21 +239,21 @@ private[http] class HttpResponseRendererFactory(
                 }
 
               case HttpEntity.Default(_, contentLength, data) ⇒
-                renderHeaders(headers.toList)
+                renderHeaders(headers)
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/rendering/HttpResponseRendererFactory.scala]: 👍

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

**Pull request merged**